### PR TITLE
add opencv_apps, proposed in #40

### PR DESCRIPTION
**Pull request by k-okada**
#40
- package names opencv_apps? opencv_examples?
- this package contains a lot of new message, show we create opencv_msgs that holds opencv basic data structure ? or ask other packages (geometry_msgs or others ) to add new messages?

here is how to run program, if you add `_debug_view` you can see debug view

`rosrun opencv_apps lk_flow _debug_view:=true image:=/image_raw`

### CODE REVIEW

**Comment by garaemon** [comment]
great, +1

File: opencv_apps/cfg/CamShift.cfg [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#  * Neither the name of Willow Garage, Inc. nor the names of its
```
Comment (vrabaud) [inline opencv_apps/cfg/CamShift.cfg:17]: hi please make sure it's your name here and not Willow garage.

File: opencv_apps/src/nodelet/convex_hull_nodelet.cpp [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+ * @author OpenCV team
+ */
+
+#include <ros/ros.h>
+#include <nodelet/nodelet.h>
+#include <image_transport/image_transport.h>
+#include <sensor_msgs/image_encodings.h>
+#include <cv_bridge/cv_bridge.h>
+
+#include "opencv2/highgui/highgui.hpp"
```
Comment (vrabaud) [inline opencv_apps/src/nodelet/convex_hull_nodelet.cpp]: please have the headers be included as

```
<opencv2/highgui/highgui.hpp>
```

**Comment by vrabaud** [comment]
wow, that is a really nice contribution ! Fix the tiny remarks and I'll merge that. I'll also make sure it works with OpenCV3.

**Comment by k-okada** [comment]
Ok , I fixed license and includes, which version of OpenCV3 are you using? opencv-3.0.0-beta https://github.com/Itseez/opencv/archive/3.0.0-beta.zip?

**Comment by vrabaud** [comment]
I still see Willow Garage in all license files. And for OpenCV3, no worries, this can wait. There is a package in indigo called OpenCV3 and it just requires some ifdef: https://github.com/ros-perception/vision_opencv/commit/8216fb5df7eb262601f12ac4b0c9415477717514

**Comment by vrabaud** [comment]
ok, merging as is, I will fix those details. Thx !

**Comment by k-okada** [comment]
Oh, sorry I just updated,

## 

◉ Kei Okada

**Comment by k-okada** [comment]
I removed all Willow Garage, https://github.com/k-okada/vision_opencv/tree/opencv_apps/opencv_apps,

**Comment by vrabaud** [comment]
ok, getting your fix in. Thx !

**Comment by vrabaud** [comment]
and the code compiles under OpenCV3: https://github.com/ros-perception/vision_opencv/commit/89a933aef7c11acdb75a17c46bfcb60389b25baa

**Pull request merged**